packages/charts/react-charting/src/utilities/utilities.ts

```diff
@@ -1216,17 +1216,23 @@ export function domainRageOfVerticalNumeric(
  * @param {ILineChartPoints[]} points
  * @returns {{ startValue: number; endValue: number }}
  */
-export function findNumericMinMaxOfY(points: ILineChartPoints[]): { startValue: number; endValue: number } {
-  const yMax = d3Max(points, (point: ILineChartPoints) => {
-    return d3Max(point.data, (item: ILineChartDataPoint) => item.y)!;
-  })!;
-  const yMin = d3Min(points, (point: ILineChartPoints) => {
-    return d3Min(point.data, (item: ILineChartDataPoint) => item.y)!;
-  })!;
+export function findNumericMinMaxOfY(
+  points: ILineChartPoints[],
+  yAxisType?: YAxisType,
+  useSecondaryYScale?: boolean,
+): { startValue: number; endValue: number } {
+  const values: number[] = [];
+  points.forEach(point => {
+    if (!useSecondaryYScale === !point.useSecondaryYScale) {
+      point.data.forEach(data => {
+        values.push(data.y);
+      });
+    }
+  });
```

Comment:
This code seems to be repeating in few places. Possible to improve without affecting readability.

Reply:
Yes, but it's difficult to extract it into a common util because each chart has a different data shape.